### DISCUSSION TITLE
BP 109 - Dédoublonner systématiquement les fichiers avant envoi

### DISCUSSION THREAD

**Discussion by Altheachevalier** [cnumr/best-practices#191] [Bonnes pratiques] [answered]
Hello, 

Concernant cette bonne pratique, je la trouve très cohérente néanmoins le titre n'est selon moi pas explicite concernant le sujet. 
On pourrait peut-être écrire "S'assurer du non dedoublonnage avant envoi"? 

Excellente journée

**Comment by ldesgrange** [comment]
Je ne suis pas certain de la pertinence de cette BP.
La plupart des entreprises (je crois avoir vu une stat de genre 80 % mais je ne la retrouve pas) utilisent un service spécialisé (mailjet, etc.) qui de base n'autorisera pas d'envoyer la même newsletter plusieurs fois au même destinataire.
Et si c'est un dev custom qui envoi des e-mails, ça relève plutôt du bug.

**Reply by ACTLEM** [reply]
Effectivement, cette mauvaise pratique semble désormais marginale. Plutôt qu'un bug, je dirais un oubli de bon sens ;-) (en soit le service est servi  correctement donc ce n'est pas véritablement une anomalie).
Ce qui m'embête surtout ici c'est la référence à l'étude de l'ADEME qui est obsolète et datée. 
En réalité, je pencherais plus sur une BP sur les newsletters pour réduire le nombre de destinataire et la limitation de pièces jointes.

**Reply by Altheachevalier** [reply]
Je suis totalement d'accord pour la BP sur la newsletter

**Reply by davidgroult** [reply]
Tout à fait d'accord également ! Comme le propose @ACTLEM, une BP sur les newsletter semble plus cohérente !

**Reply by arnolem** [reply]
Je ne peux que être d'accord avec vous 😅
https://twitter.com/arnolem/status/1444768409045057544

En plus de ce que propose @ACTLEM, j'ajouterais de limiter les envois d'email transactionnel.
Il n'est pas rare de recevoir un mail par action dans des applications de gestion de tâche, ou plusieurs mails lors d'un achat sur internet (Création de compte, commande, confirmation de paiement, expédition, satisfaction, relance, satisfaction livraison, facture, ...)

**Reply by devcrafting** [reply]
Je mettrai aussi l'emphase sur la limitation des pièces jointes, il vaut mieux envoyer un lien de téléchargement, non?
Je me suis battu assez tôt pour arrêter l'envoi de fichiers joints et plutôt mettre des liens vers les fichiers sur des partages réseaux/drive (surtout pour le versioning du fichier à la base)

**Reply by Altheachevalier** [reply]
Je suis totalement d'accord avec le fait de mentionner les emails transactionnels et les PJ. 

Que pensez-vous de faire une BP générale sur l'utilisation des emails ?  (newsletters, email transactionnel et PJ)

**Reply by florinesueur** [reply]
Il y a eu un point sur la discussion #20.

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé une issue pour modifier cette BP plus autour des newsletters: https://github.com/cnumr/best-practices/issues/272

@Altheachevalier je te l'ai affectée mais si tu préfères que je la prennes en charge pas de problème.
